pkg/controller/datavolume-controller.go

```diff
@@ -497,12 +497,19 @@ func (r *DatavolumeReconciler) reconcileClone(log logr.Logger,
 
 	if !prePopulated && !pvcPopulated {
 		if pvc == nil {
+			storageClass, err := GetStorageClassByName(r.client, pvcSpec.StorageClassName)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if storageClass == nil {
+				return reconcile.Result{}, fmt.Errorf("Could not find a default storage class in the cluster")
+			}
 			if selectedCloneStrategy == SmartClone {
-				snapshotClassName, _ := r.getSnapshotClassForSmartClone(datavolume, pvcSpec)
+				snapshotClassName, _ := r.getSnapshotClassForSmartClone(datavolume, storageClass)
 				return r.reconcileSmartClonePvc(log, datavolume, pvcSpec, transferName, snapshotClassName)
 			}
 			if selectedCloneStrategy == CsiClone {
```

Comment:
In theory we should not be able to get SmartClone is there is no storageClass/defaultStoratgeClass, so it should not get here. 
Anyway, there is an error we ignore here. Maybe we should not?

Reply:
Also, maybe we should look into the `selectCloneStrategy`  and revise what is returned in case there is no StorageClass.

Reply:
Yeah I think we shouldn't have ignored the error.
I minimized the scope of this commit to handle the nils without altering logic too much

Reply:
@brybacki Should be strictly about avoiding nils now. let me know what you think